assets/src/components/with-term/index.js

```diff
@@ -33,30 +33,66 @@ export default function withTerm() {
 			const [ term, setTerm ] = useState( null );
 
 			useEffect( () => {
-				createTerm( taxonomy, label, value )
-					.catch( error => {
-						if ( error.code !== 'term_exists' ) {
-							return Promise.reject( error );
+				function handleSearchTerms( terms ) {
+					if ( terms.length ) {
+						setTerm( terms[ 0 ] );
+					}
+				}
+
+				function handleError( error ) {
+					if ( error.code !== 'term_exists' ) {
+						return Promise.reject( error );
+					}
+
+					return searchTerms( taxonomy, value ).then( searchTermsResponse => {
+						if ( searchTermsResponse instanceof Response ) {
+							searchTermsResponse.json().then(
+								terms => {
+									handleSearchTerms( terms );
+								}
+							);
+						} else {
+							handleSearchTerms( searchTermsResponse );
 						}
+					} );
+				}
+
+				function handeNewTerm( newTerm ) {
+					const { id, name, slug } = newTerm;
 
-						return searchTerms( taxonomy, value ).then( terms => {
-							if ( terms.length ) {
-								setTerm( terms[ 0 ] );
-							}
-						} );
+					setTerm( {
+						id,
+						name,
+						slug,
+					} );
+				}
+
+				createTerm( taxonomy, label, value )
+					.catch( response => {
+						if ( response instanceof Response ) {
+							response.json().then(
+								error => {
+									handleError( error );
+								}
+							);
+						} else {
+							handleError( response );
+						}
 					} )
-					.then( newTerm => {
-						if ( ! newTerm ) {
+					.then( response => {
+						if ( ! response ) {
 							return;
 						}
 
-						const { id, name, slug } = newTerm;
-
-						setTerm( {
-							id,
-							name,
-							slug,
-						} );
+						if ( response instanceof Response ) {
+							response.json().then(
+								newTerm => {
+									handeNewTerm( newTerm );
```

Comment:
```js
							response.json().then( handleNewTerm );
```
Since the callback does nothing but pass the argument to the handler, the handler could be used itself in place of the callback